warehouse/query-core/src/main/java/datawave/core/iterators/GlobalIndexTermMatchingIterator.java

```diff
@@ -87,6 +87,17 @@ public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean i
         this.scanCFs = columnFamilies;
         this.scanInclusive = inclusive;
         getSource().seek(range, columnFamilies, inclusive);
+        
+        // if we have been reseeked after being torn down, and we are only returning unique terms, then advance to the next unique row/cf
+        if (!range.isStartKeyInclusive() && uniqeTermsOnly && getSource().hasTop()) {
```

Comment:
why are uniqueTerms a condition of a rebuild?

Reply:
If we are not dealing with uniqueTerms, then we can simple continue scanning per the supplied range to produce the correct results.  Only when dealing with uniqueTerms will we need to advance forward on a rebuild.